### DISCUSSION TITLE
Where do syntax errors get flagged in the API?

### DISCUSSION THREAD

**Discussion by kcameron4456** [MikePopoloski/slang#730] [General]
The way I'm using the parser, it seems to accept almost anything without generating an error message.
Here's a simple example:

```#include <slang/syntax/SyntaxTree.h>
#include <slang/syntax/SyntaxNode.h>

#include <iostream>

using namespace std;
using namespace slang;
using namespace slang::syntax;

int main (int argc, char **argv) {
    auto tree = SyntaxTree::fromText("module A; 12 ee3 3243234  endmodule\n");
    SyntaxNode &Node = tree->root();
    cout << "root node = " << Node.toString() << endl;

    return 0;
}
~                                                                                                                                                                                             
```

This produces the output:
root node = module A; ee3  endmodule

How do I get slang to fully enforce SystemVerilog syntax rules?

**Comment by MikePopoloski** [comment]
You can iterate over the diagnostics returned by the [diagnostics()](https://sv-lang.com/classslang_1_1syntax_1_1_syntax_tree.html#a5afdd762c70295225497bc47e569711d) function
